### PR TITLE
feat(#449): add swap space by default

### LINKED ISSUE #449: Infra: Add swap space to hosts
## What

We need to add a small swap partition to hosts, especially needed for our hosts with minimal memory footprint.

1G is enough

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Adds swap with configurability via pillar datum

https://serverfault.com/a/865797

## Closes

- Closes #449

### CODE REVIEW

**Comment by JacobCoffee** [comment]
Currently blocking consul migration

File: salt/base/swap.sls [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    - name: |
+        swapon --show=NAME --noheadings | grep -q "^{{ path }}$" && swapoff {{ path }}
+        rm -f {{ path }}
+        fallocate -l {{ size }}M {{ path }}
+        chmod 0600 {{ path }}
+        mkswap {{ path }}
+    - unless: bash -c "[[ $(($(stat -c %s {{ path }}) / 1024**2)) = {{ size }} ]]"
+
+  mount.swap:
+    - persist: true
```
Comment (JacobCoffee) [inline salt/base/swap.sls:16]: we could also add to fstab if that is better

File: salt/base/swap.sls [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{% set swapfile = salt["pillar.get"]("swapfile", {}) %}
+{% set size = swapfile.get("size", "1024") %}
+{% set path = swapfile.get("path", "/swapfile") %}
```
Comment (ewdurbin) [inline salt/base/swap.sls:3]:
```suggestion
{% set path = swapfile.get("path") %}
```

I'm not sure want to _always_ add this (really thinking mostly in dev where it is almost never necessary), so give a way to opt out, see additional suggested change.

File: salt/base/swap.sls [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+{{ path }}:
```
Comment (ewdurbin) [inline salt/base/swap.sls:5]: Given suggested change above, wrapping this in a `{% if path is not None %}` would make this opt-in.

**Review by ewdurbin — APPROVED** [review]

File: salt/base/swap.sls (hunk 1 above)
Comment (ewdurbin) [reply salt/base/swap.sls:16]: this seems fine!

**Pull request merged**